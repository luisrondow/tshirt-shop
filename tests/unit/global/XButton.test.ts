import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'

import { XButton } from '#components'

describe('XButton component', () => {
  it('should render correctly', () => {
    const wrapper = mount(XButton)

    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should call "onClick" event', () => {
    const onClick = vi.fn()
    const wrapper = mount(XButton, {
      props: {
        onClick,
      },
    })

    wrapper.trigger('click')

    expect(onClick).toHaveBeenCalled()
  })

  it('should not call "onClick" event if is disabled', () => {
    const onClick = vi.fn()
    const wrapper = mount(XButton, {
      props: {
        onClick,
        disabled: true,
      },
    })

    wrapper.trigger('click')

    expect(onClick).not.toHaveBeenCalled()
  })
})
