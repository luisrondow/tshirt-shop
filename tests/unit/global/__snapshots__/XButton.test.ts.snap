// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`XButton component > should render correctly 1`] = `
<button
  class="h-10 w-full uppercase border border-black bg-black px-4 py-2 text-white  hover:bg-white hover:text-black"
>
  
  
</button>
`;
